text/src/swam/witx/parser/TypesParser.scala

```diff
@@ -0,0 +1,102 @@
+package swam
+package witx
+package parser
+
+import fastparse._
+import swam.witx.parser.parser.WitxWhitespace._
+import swam.witx.unresolved._
+
+import scala.collection.MapView
+
+/**
+    @author Javier Cabrera-Arteaga on 2020-03-18
+  */
+object TypesParser {
+
+  import Lexical._
+
+  private var declaredTypes = Map[String, BaseWitxType](
```

Comment:
This is really dangerous. Please use immutability everywhere, especially in an object.